### ISSUE TITLE
API suggestions

### ISSUE THREAD

**Issue by johnrutherford** [JuergenGutsch/graphql-aspnetcore#8]
I have some suggested improvements to the API. Let me know if you're open to them, and I'd be happy to make the changes.

I think something like this would be better for configuring the middleware:

```
app.UseGraphQl<BooksQuery>("/graph-api", options =>
{
    options.AuthorizationPolicy = "AuthenticatedUser";
    options.FormatOutput = false;
});
```

This has the following advantages:

1. The root graph type is passed as a type parameter. This makes the code look cleaner, but more importantly, will allow the middleware to resolve the type from the container on each request. Currently all dependencies will be singletons.
2. I think moving the path configuration to the method call makes it more obvious. It also allows you to add the middleware to the pipeline using Map(), so you don't need to check the path within the middleware itself. 
3. Finally, adding an `AuthorizationPolicy` property allows one to apply a defined authorization policy to the endpoint.

**Comment by JuergenGutsch** [comment]
Looks awesome. I just hacked it down to get it running :)

You need to pass the repository somewhere. This is why I created the instance of the BooksQuery directly and pass it via the options. Maybe you have an idea about that.

Feel free to create a PR. Thanks :)
(I use [GirFlow](http://danielkummer.github.io/git-flow-cheatsheet/), if you don't, please create a feature branch out of the "develop" branch and a PR afterwards pointing to "develop")

We should also change the API for GraphiQL a little bit:
```
app.UseGraphiql("/graphiql", options =>
{
  options.GraphQlEndpoint = "/graph"; // default
});
```

**Comment by johnrutherford** [comment]
I just created a pull request. I ended up making a lot more changes that originally planned.

**Comment by JuergenGutsch** [comment]
Great Thanks 👍 Will have a look asap

**Comment by amborle** [comment]
Did you guys get anywhere with this proposal?

For my own use cases, I really need to resolve my dependencies on each request (point 1 made by @johnrutherford above) - so singleton dependencies are not enough for me.

**Comment by JuergenGutsch** [comment]
@amborle yes, the proposal is merged (#9) into the develop branch.
